samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/integration/helper/integrated/AssessmentGradeInfoProvider.java

```diff
@@ -85,28 +87,27 @@ public String getAppKey() {
 
     
     private PublishedAssessmentIfc getPublishedAssessment(String id) {
-        PublishedAssessmentIfc a = null;
-        if (pubAssessmentCache.containsKey(id)) {
-            a = (PublishedAssessmentIfc) pubAssessmentCache.get(id);
-            if(a != null) {
-                if (log.isDebugEnabled()) {
-                    log.debug("returning assesment " + id + " from cache");
-                }
-                return a;
-            }
-            /* Below we may fail to re-establish the value */
-            pubAssessmentCache.remove(id);
-    	}
+        // SAM-3068 avoid looking up another tool's id
+        if (!StringUtils.isNumeric(id)) {
+            return null;
+        }
+
+        PublishedAssessmentIfc a = (PublishedAssessmentIfc) pubAssessmentCache.get(id);
```

Comment:
Okay this change looks good, it looks like isNumeric will also filter out an id of null which seems like it may or may not have been cached before too.